prov/coll/src/coll_coll.c

```diff
@@ -941,11 +951,14 @@ int coll_join_collective(struct fid_ep *ep, const void *addr,
 	if (!new_coll_mc)
 		return -FI_ENOMEM;
 
+	if (flags & FI_PEER)
+		new_coll_mc->peer_mc = context;
+
 	/* get the rank */
 	coll_find_local_rank(ep, new_coll_mc);
 	coll_find_local_rank(ep, coll_mc);
 
-	join_op = coll_create_op(ep, coll_mc, UTIL_COLL_JOIN_OP, flags,
+	join_op = coll_create_op(ep, new_coll_mc, UTIL_COLL_JOIN_OP, flags,
```

Comment:
This changes how join_op works. It is supposed to work on the original coll_mc and the new coll_mc is specified as a parameter of the join_op. See L955 (or L968 after patch).

Reply:
Roll-back